### DISCUSSION TITLE
Deploy Magma on Windows machine

### DISCUSSION THREAD

**Discussion by robinge330** [magma/magma#5372] [Q&A]
Hi,
 i have the
1. pre-requisite of  Virtual Box, Vagrant and Ansible deployed.
2. Cloned the magma repository from using git clone git@github.com:magma/magma.git
3.  now from command prompt C:\Users\user\.ssh\magma\lte\gateway on executing the vagrant up magma
error is 

_*_**_*Vagrant failed to initialize at a very early stage:
The plugins failed to initialize correctly. This may be due to manual
modifications made within the Vagrant home directory. Vagrant can
attempt to automatically correct this issue by running:

  vagrant plugin repair

If Vagrant was recently updated, this error may be due to incompatible
versions of dependencies. To fix this problem please remove and re-install
all plugins. Vagrant can attempt to do this automatically by running:

  vagrant plugin expunge --reinstall

Or you may want to try updating the installed plugins to their latest
versions:

  vagrant plugin update

Error message given during initialization: Is a directory @ rb_sysopen - C

Please could i get a guidance

**Comment by hcgatewood** [comment]
@pshelar @tmdzk any thoughts?

**Reply by tmdzk** [reply]
`C:\Users\u...` that smells like a Windows. 
We don't support Windows development. (https://docs.magmacore.org/docs/basics/prerequisites#development-tools)

However from what I see It's just missing a couple of plugin. I'd try executing `vagrant plugin expunge --reinstall` and `vagrant plugin update`

**Reply by robinge330** [reply]
vagrant plugin update gives error as

         1: from C:/HashiCorp/Vagrant/embedded/mingw64/lib/ruby/2.6.0/rubygems/config_file.rb:354:in `load_file'
C:/HashiCorp/Vagrant/embedded/mingw64/lib/ruby/2.6.0/rubygems/config_file.rb:354:in `read': Is a directory @ rb_sysopen - C (Errno::EISDIR)

and on rerun of vagrant up magma

Error message given during initialization: Is a directory @ rb_sysopen - C


Does it mean that magma deployment on laptop is not supported ?

**Reply by hcgatewood** [reply]
We currently don't actively support deployment on Windows machines. For now, you can either

- Switch to macOS or Linux machine
- Consider contributing to the project by adding support for dev deployment on Windows machines 🙂

**Reply by robinge330** [reply]
Thanks for guidance

The below document did confuse me, so i thought the support still exists.
https://github.com/magma/magma/blob/master/docs/Magma_Deployment_Guide_Installing_Magma_V2.pdf

will try on Linux machine.
Do we support AGW deployment on VM's or it has to be on baremetal
